packages/components/src/internal/components/chart/ChartSettingsPanel.tsx

```diff
@@ -330,17 +329,37 @@ export const ChartSettingsPanel: FC<Props> = memo(props => {
 
     const legendOptions = useMemo(() => {
         return [
-            { label: 'Right', selected: !legendPos || legendPos === 'right', value: 'right' },
-            { label: 'Bottom', selected: legendPos === 'bottom', value: 'bottom' },
+            { label: 'Right', selected: !chartConfig.legendPos || chartConfig.legendPos === 'right', value: 'right' },
+            { label: 'Bottom', selected: chartConfig.legendPos === 'bottom', value: 'bottom' },
         ];
-    }, [legendPos]);
+    }, [chartConfig.legendPos]);
 
     const onLegendPosChange = useCallback(
         value => setChartConfig(current => ({ ...current, legendPos: value })),
         [setChartConfig]
     );
 
+    const hideDataPointsOptions = useMemo(() => {
+        return [
+            {
+                label: 'Show',
+                selected:
+                    chartConfig.geomOptions.hideDataPoints === undefined ||
+                    chartConfig.geomOptions.hideDataPoints === false,
+                value: 'false',
+            },
+            { label: 'Hide', selected: chartConfig.geomOptions.hideDataPoints === true, value: 'true' },
+        ];
+    }, [chartConfig.geomOptions.hideDataPoints]);
```

Comment:
No need for a return if you wrap the body in parens instead of curlies.
```suggestion
    const hideDataPointsOptions = useMemo(
        () => [
            {
                label: 'Show',
                selected:
                    chartConfig.geomOptions.hideDataPoints === undefined ||
                    chartConfig.geomOptions.hideDataPoints === false,
                value: 'false',
            },
            { label: 'Hide', selected: chartConfig.geomOptions.hideDataPoints === true, value: 'true' },
        ],
        [chartConfig.geomOptions.hideDataPoints]
    );
```

Reply:
done